python/lsst/cp/verify/mergeResults.py

```diff
@@ -0,0 +1,323 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.pex.config as pexConfig
+
+
+__all__ = ['CpVerifyExpMergeConfig', 'CpVerifyExpMergeTask',
+           'CpVerifyRunMergeConfig', 'CpVerifyRunMergeTask']
+
+
+class CpVerifyExpMergeConnections(pipeBase.PipelineTaskConnections,
+                                  dimensions={"instrument", "exposure"},
+                                  defaultTemplates={}):
+    inputStats = cT.Input(
+        name="detectorStats",
+        doc="Input statistics to merge.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "exposure", "detector"],
+        multiple=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        storageClass="Camera",
+        doc="Input camera.",
+        dimensions=["instrument", ],
+        isCalibration=True,
+    )
+
+    outputStats = cT.Output(
+        name="exposureStats",
+        doc="Output statistics.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "exposure"],
+    )
+
+
+class CpVerifyExpMergeConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CpVerifyExpMergeConnections):
+    """Configuration paramters for exposure stats merging.
+    """
+    exposureStatKeywords = pexConfig.DictField(
+        keytype=str,
+        itemtype=str,
+        doc="Statistics to run on the set of exposure values.",
+        default={},
+    )
+
+
+class CpVerifyExpMergeTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    """Merge statistics from detectors together.
+    """
+    ConfigClass = CpVerifyExpMergeConfig
+    _DefaultName = 'cpVerifyExpMerge'
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        dimensions = [exp.dataId.byName() for exp in inputRefs.inputStats]
+        inputs['inputDims'] = dimensions
+
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, inputStats, camera, inputDims):
+        """Merge statistics.
+
+        Parameters
+        ----------
+        inputStats : `list` [`dict`]
+            Measured statistics for a detector.
+        camera : `lsst.afw.cameraGeom.Camera`
+            The camera geometry for this exposure.
+        inputDims : `list` [`dict`]
+            List of dictionaries of input data dimensions/values.
+            Each list entry should contain:
+
+            ``"exposure"``
+                exposure id value (`int`)
+            ``"detector"``
+                detector id value (`int`)
+
+        Returns
+        -------
+        outputStats : `dict`
+            Merged full exposure statistics.
+
+        Notes
+        -----
+        The outputStats should have a yaml representation of the form:
```

Comment:
Is it not possible to make this a dataclass with a `write_yaml` method or something? It would add a little more boilerplate but is much more easily documented than multiply-nested dicts.

Reply:
The short answer is yes, the long answer is that I've added a ticket to do that: https://jira.lsstcorp.org/browse/DM-29643